From 89744d9daffd60c69f1a0bdcf512ecc87c451753 Mon Sep 17 00:00:00 2001
From: Brandon Maier <brandon.maier@gmail.com>
Date: Tue, 18 Feb 2025 11:23:50 -0600
Subject: [PATCH 4/4] GNUmakefile: fix install of `ln` to $PATH

The following error occurs when packaging uutils for conda-forge and
targeting a cross-compiled architecture.

  cd $PREFIX/bin && ln -fs coreutils ln
  cd $PREFIX/bin && ln -fs coreutils logname
  /bin/sh: $PREFIX/bin/ln: Bad CPU type in executable

This is because the uutils `ln` gets installed to a directory in $PATH.
But it's cross-compiled for a different architecture and so can't be
run.
---
 GNUmakefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/GNUmakefile b/GNUmakefile
index eb8d8d13f..e6149d7e7 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -22,6 +22,7 @@ ifeq ($(PROFILE),release)
 endif
 
 RM := rm -rf
+LN := $(shell command -v ln 2> /dev/null)
 
 # Binaries
 CARGO  ?= cargo
@@ -388,9 +389,9 @@ install: build install-manpages install-completions
 ifeq (${MULTICALL}, y)
 	$(INSTALL) $(BUILDDIR)/coreutils $(INSTALLDIR_BIN)/$(PROG_PREFIX)coreutils
 	$(foreach prog, $(filter-out coreutils, $(INSTALLEES)), \
-		cd $(INSTALLDIR_BIN) && ln -fs $(PROG_PREFIX)coreutils $(PROG_PREFIX)$(prog) $(newline) \
+		cd $(INSTALLDIR_BIN) && $(LN) -fs $(PROG_PREFIX)coreutils $(PROG_PREFIX)$(prog) $(newline) \
 	)
-	$(if $(findstring test,$(INSTALLEES)), cd $(INSTALLDIR_BIN) && ln -fs $(PROG_PREFIX)coreutils $(PROG_PREFIX)[)
+	$(if $(findstring test,$(INSTALLEES)), cd $(INSTALLDIR_BIN) && $(LN) -fs $(PROG_PREFIX)coreutils $(PROG_PREFIX)[)
 else
 	$(foreach prog, $(INSTALLEES), \
 		$(INSTALL) $(BUILDDIR)/$(prog) $(INSTALLDIR_BIN)/$(PROG_PREFIX)$(prog) $(newline) \
-- 
2.48.1

